Automattic/jetpack
# SSO: simplify the logic when inviting new users to WordPress.com.

## Proposed changes:

This should help further editing this method in the future:

- It splits the different reasons an invite should error out into separate conditions.
- It adds extra checks before an invite is sent.
- It adds extra sanitization on the different strings being used.
- It adds an explicit error message when there are connection issues on the site.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Go to Jetpack > Settings > Security 
* Enable the SSO feature
* Go to Users > Add New
* Fill in the fields, including the checkbox to invite the user to WordPress.com and the custom message.
* Add the user
    * Ensure the invite email is sent with the custom message.
